contracts/QuestFactory.sol

```diff
@@ -231,6 +232,9 @@ contract QuestFactory is Initializable, OwnableUpgradeable, AccessControlUpgrade
     function mintReceipt(string memory questId_, bytes32 hash_, bytes memory signature_) public {
         if (quests[questId_].numberMinted + 1 > quests[questId_].totalParticipants) revert OverMaxAllowedToMint();
         if (quests[questId_].addressMinted[msg.sender] == true) revert AddressAlreadyMinted();
+        if (!QuestContract(quests[questId_].questAddress).hasStarted()) revert QuestNotStarted();
+        if (block.timestamp < QuestContract(quests[questId_].questAddress).startTime()) revert QuestNotStarted();
+        if (block.timestamp > QuestContract(quests[questId_].questAddress).endTime()) revert QuestEnded();
```

Comment:
should we just create a struct on the quest contract that has these variables and return the struct here so we don't have to make 3 calls?

Reply:
Just following up here, will keep as is as it doesn't really save on gas to make these calls